### PR TITLE
Manipulate graph targets and implement New Graph -> New Empty Graph

### PR DESCRIPTION
**Pull request by cbowman0**
Buttons to re-order and remove graph targets on the Graph Target grid.

Add a new target to a graph.

Up and Down buttons are from http://tango.freedesktop.org/.  Delete button already existed in the source.

Add New Graph -> Empty Graph.

### CODE REVIEW

**Comment by cbowman0** [comment]
For @obfuscurity: 
maniptargets

File: webapp/content/js/dashboard.js [hunk 1]
```diff
@@ -1952,6 +2043,24 @@ function graphClicked(graphView, graphIndex, element, evt) {
...
+      width: buttonWidth,
+      handler: function() {
+                 // Hide the other menus
+                 operationsMenu.hide();
+                 optionsMenu.doHide(); // private method... yuck
+                 functionsMenu.hide();
+
+                 targetStore.add([ new targetStore.recordType({target: 'Edit to save'}) ]);
+                 targets.push('Edit to save');
+                 targetGrid.setHeight((rowHeight * Math.min(targets.length, maxRows)) + frameHeight);
```
Comment (obfuscurity) [inline webapp/content/js/dashboard.js]: Does the indention look off here or is it my imagination?

**Comment by cbowman0** [comment]
Looks like it's 4 space instead of 2 space.  I'll correct it.

**Comment by cbowman0** [comment]
Can I get a peer review of this?

**Comment by obfuscurity** [comment]
I'll try to take a look at the code later tonight. Initial UX thoughts: the navigation elements (green arrows) are different than the approach taken in the Composer (e.g. Move > Move Up). I think I like your approach a little better anyways, so NBD. But we probably shouldn't allow illegal choices (e.g. arrow up for the top element) unless that implies a wraparound action to the list.

**Comment by obfuscurity** [comment]
@cbowman0 Sorry for the delay, looking into this now.

**Comment by obfuscurity** [comment]
@cbowman0 Played around with it a bit. I think the feature would be much more useful if the new target field supported autocompletion. Without it, the feature feels not altogether fleshed out imho.

Is this something you'd be willing to take a stab at?

**Comment by cbowman0** [comment]
Do you mean in the pop-up after clicking a graph, choosing new target and it adds a blank target line?

**Comment by obfuscurity** [comment]
Correct.

**Comment by cbowman0** [comment]
I'm not sure where to start with adding auto-completion to the add new target box.  Do others believe this is required to merge this in?

**Comment by cbowman0** [comment]
@DanCech or @deniszh, any objections to merging this as is?

**Comment by DanCech** [comment]
It's definitely an improvement, also not sure where to start with auto-complete as I haven't done any dev on the graphite-web ui.  I'd suggest opening an issue for adding auto-complete to it, then going ahead with the merge.

**Comment by cbowman0** [comment]
#2097 created.  Merging now.

**Pull request merged**